src/Control/Effect/Parse.hs

```diff
@@ -28,11 +32,11 @@ instance HFunctor Parse where
   hmap f (Parse parser blob k) = Parse parser blob (f . k)
 
 instance Effect Parse where
-  handle state handler (Parse parser blob k) = Parse parser blob (handler . (<$ state) . k)
+  thread state handler (Parse parser blob k) = Parse parser blob (handler . (<$ state) . k)
```

Comment:
`handle` has been renamed to `thread`, as the method doesn’t really handle anything, but rather threads a handler through the higher-order positions & continuation in the effect.